--- pyiboot01_bootstrap.py	2019-08-12 17:37:25.460219000 +1000
+++ pyiboot01_bootstrap.py	2019-08-13 18:10:32.084968800 +1000
@@ -123,11 +123,12 @@
 if sys.warnoptions:
     import warnings
 
-try:
+def _setup_ctypes():
     import ctypes
     import os
+    import sys
     from ctypes import LibraryLoader, DEFAULT_MODE
-
+    
     def _frozen_name(name):
         if name:
             frozen_name = os.path.join(sys._MEIPASS, os.path.basename(name))
@@ -186,6 +186,8 @@
 
         ctypes.OleDLL = PyInstallerOleDLL
         ctypes.oledll = LibraryLoader(PyInstallerOleDLL)
+try:
+    _setup_ctypes()
 except ImportError:
     pass
 
@@ -213,4 +215,4 @@
 # use-case, see issue #653.
 if os.path.isdir(d):
     for fn in os.listdir(d):
-        sys.path.append(os.path.join(d, fn))
+        sys.path.append(os.path.join(d, fn)) 
\ No newline at end of file
